zproject/backends.py

```diff
@@ -917,17 +917,28 @@ def social_associate_user_helper(backend: BaseAuth, return_data: Dict[str, Any],
 
         if not chosen_email:
             avatars = {}  # Dict[str, str]
+            existing_account_emails = []
             for email in verified_emails:
                 existing_account = common_get_active_user(email, realm, {})
                 if existing_account is not None:
+                    existing_account_emails.append(email)
                     avatars[email] = avatar_url(existing_account)
-
-            return render(backend.strategy.request, 'zerver/social_auth_select_email.html', context = {
-                'primary_email': verified_emails[0],
-                'verified_non_primary_emails': verified_emails[1:],
-                'backend': 'github',
-                'avatar_urls': avatars,
+            if(len(existing_account_emails) > 1):
+                return render(backend.strategy.request, 'zerver/social_auth_select_email.html', context = {
+                    'primary_email': verified_emails[0],
+                    'verified_non_primary_emails': verified_emails[1:],
+                    'backend': 'github',
+                    'avatar_urls': avatars,
+            })
+            elif(backend.strategy.session_get('is_signup') == '1'):
```

Comment:
Should this be the way it is or ``backend.strategy.request_data().get('is_signup')`` (that's how we grab the email a bit above)? Perhaps the effect is the same, but I'm not sure about it, we should figure that out.

Reply:
I thought it should be this way because `is_signup = strategy.session_get('is_signup') == '1'` was used in `social_auth_finish` function below.

Reply:
Ah, you're right. ``is_signup`` gets saved in the session because it's in ``settings.SOCIAL_AUTH_FIELDS_STORED_IN_SESSION``. So this should be the correct way indeed.